org.lflang/src/org/lflang/generator/JavaGeneratorUtils.java

```diff
@@ -312,13 +313,20 @@ public static LFResource getLFResource(
      * @param text The text to be written.
      * @param path The file to write the code to.
      */
-    public static void writeToFile(CharSequence text, String path) throws IOException {
-        new File(path).getParentFile().mkdirs();
-        try (BufferedWriter writer = new BufferedWriter(new FileWriter(path))) {
-            for (int i = 0; i < text.length(); i++) {
-                writer.write(text.charAt(i));
-            }
+    public static void writeToFile(String text, Path path) throws IOException {
+        if (!path.getParent().toFile().mkdirs()) {
+            throw new IOException("Could not create the directory " + path);
         }
+        Files.write(path, text.getBytes());
+    }
+
+    /**
+     * Write text to a file.
+     * @param text The text to be written.
+     * @param path The file to write the code to.
+     */
+    public static void writeToFile(CharSequence text, Path path) throws IOException {
```

Comment:
(For context, the motivation for using `CharSequence` here instead of `String` was that the C generator was having out-of-memory errors because of its large strings, and `toString()` does create a copy. However, I guess that is not important anymore since the C generator has been repaired by `scalability-banks`.)